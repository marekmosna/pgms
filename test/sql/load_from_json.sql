\set ECHO none
BEGIN;

\i test/pgtap-core.sql
\i sql/pgms.sql

SELECT plan(4);

SELECT is_empty(
    $$ SELECT FROM load_from_json(NULL) AS (s spectrum) $$,
    'json parser should deal with NULL json'
);

SELECT row_eq(
    $$ SELECT FROM load_from_json('$$||v||$$'::jsonb) AS (s spectrum) $$,
    ROW(),
    'json parser should deal with empty json'
)FROM unnest(ARRAY[
   '{}',
   '[]'
]) AS v;

SELECT results_eq(
    $$ SELECT "peaks_json" FROM load_from_json('$$||v||$$'::jsonb) AS ("peaks_json" spectrum) $$,
    $$ VALUES('{ {189.123456789, 189.00, -189, 189.48956}, {1.9, 1.9, -1.9, 0} }'::spectrum) $$,
    'json parser should deal with sign fixures'
)FROM unnest(ARRAY[
'{
  "peaks_json": [
      [
          189.123456789,
          1.9
      ],
      [
          189.00,
          1.9
      ],
      [
          -189,
          -1.9
      ],
      [
          189.48956,
          0
      ]
  ]
}'
]) AS v;

SELECT * FROM finish();
ROLLBACK;
