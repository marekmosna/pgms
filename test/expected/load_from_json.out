\set ECHO none
1..4
ok 1 - json parser should deal with NULL json
ok 2 - json parser should deal with empty json
ok 3 - json parser should deal with empty json
ok 4 - json parser should deal with sign fixures
